File: typescript/src/workshop/m3/s15_invariants/step2/BookingService.ts
import type { Decimal } from 'decimal.js';

import { Reservation } from './Reservation.js';

/** Krok 2: serwis nie powtarza reguł modelu - tylko koordynuje. */
export class BookingService {
  book(email: string, seats: number, total: Decimal): string {
    const reservation = new Reservation(email, seats, total);
    return `zarezerwowano: ${reservation.email}, miejsc ${reservation.seats}`
      + `, kwota ${reservation.total.toFixed(2)}`;
  }
}

File: csharp/src/Training.Workshop/M3/S15Invariants/Step2/BookingService.cs
using System.Globalization;

namespace Training.Workshop.M3.S15Invariants.Step2;

/// <summary>Krok 2: serwis nie powtarza reguł modelu - tylko koordynuje.</summary>
public sealed class BookingService
{
    public string Book(string email, int seats, decimal total)
    {
        var reservation = new Reservation(email, seats, total);
        return "zarezerwowano: " + reservation.Email + ", miejsc " + reservation.Seats
            + ", kwota " + reservation.Total.ToString(CultureInfo.InvariantCulture);
    }
}
